### ISSUE TITLE
Type 'HttpRequest' not found

### ISSUE THREAD

**Issue by siloebb** [dart-lang/intl#286]
I install the lib but when I try to run the app I got that error below. I think thar was I update the flutter to 1.17, I tried to clean the project and the cache.

`Launching lib/main.dart on SM N9600 in debug mode...
Running Gradle task 'assembleDebug'...

Compiler message:
../../../.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:11:8: Error: Not found: 'dart:html'
import 'dart:html';
       ^
../../../.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:28:10: Error: Type 'HttpRequest' not found.
  Future<HttpRequest> _getString(String url, HttpRequest xhr) {
         ^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:28:46: Error: Type 'HttpRequest' not found.
  Future<HttpRequest> _getString(String url, HttpRequest xhr) {
                                             ^^^^^^^^^^^
Invalid depfile: /home/siloe/Developer/flutter/meditacao/.dart_tool/flutter_build/659e08df06b3af32a6baf6232e5f9259/kernel_snapshot.d
Invalid depfile: /home/siloe/Developer/flutter/meditacao/.dart_tool/flutter_build/659e08df06b3af32a6baf6232e5f9259/kernel_snapshot.d

Compiler message:
../../../.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:11:8: Error: Not found: 'dart:html'
import 'dart:html';
       ^
../../../.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:20:19: Error: The method 'HttpRequest' isn't defined for the class 'HttpRequestDataReader'.
 - 'HttpRequestDataReader' is from 'package:intl/src/http_request_data_reader.dart' ('../../../.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart').
Try correcting the name to the name of an existing method, or defining a method named 'HttpRequest'.
    var request = HttpRequest();
                  ^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:28:46: Error: 'HttpRequest' isn't a type.
  Future<HttpRequest> _getString(String url, HttpRequest xhr) {
                                             ^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:29:31: Error: 'HttpRequest' isn't a type.
    var completer = Completer<HttpRequest>();
                              ^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:28:10: Error: Type 'HttpRequest' not found.
  Future<HttpRequest> _getString(String url, HttpRequest xhr) {
         ^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:28:46: Error: Type 'HttpRequest' not found.
  Future<HttpRequest> _getString(String url, HttpRequest xhr) {
                                             ^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:20:19: Error: The method 'HttpRequest' isn't defined for the class 'HttpRequestDataReader'.
 - 'HttpRequestDataReader' is from 'package:intl/src/http_request_data_reader.dart' ('../../../.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart').
Try correcting the name to the name of an existing method, or defining a method named 'HttpRequest'.
    var request = HttpRequest();
                  ^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:28:46: Error: 'HttpRequest' isn't a type.
  Future<HttpRequest> _getString(String url, HttpRequest xhr) {
                                             ^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:29:31: Error: 'HttpRequest' isn't a type.
    var completer = Completer<HttpRequest>();
                              ^^^^^^^^^^^
Unhandled exception:
FileSystemException(uri=org-dartlang-untranslatable-uri:dart%3Ahtml; message=StandardFileSystem only supports file:* and data:* URIs)
#0      StandardFileSystem.entityForUri (package:front_end/src/api_prototype/standard_file_system.dart:33:7)
#1      asFileUri (package:vm/kernel_front_end.dart:567:37)
<asynchronous suspension>
#2      writeDepfile (package:vm/kernel_front_end.dart:760:21)
<asynchronous suspension>
#3      FrontendCompiler.compile (package:frontend_server/frontend_server.dart:536:15)
<asynchronous suspension>
#4      _FlutterFrontendCompiler.compile (package:flutter_frontend_server/server.dart:40:22)
#5      starter (package:flutter_frontend_server/server.dart:178:27)
#6      main (file:///b/s/w/ir/cache/builder/src/flutter/flutter_frontend_server/bin/starter.dart:8:30)
#7      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#8      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)

Target kernel_snapshot failed: Exception: Errors during snapshot creation: null
build failed.

FAILURE: Build failed with an exception.

* Where:
Script '/home/siloe/flutter/packages/flutter_tools/gradle/flutter.gradle' line: 882

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command '/home/siloe/flutter/bin/flutter'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 16s
Exception: Gradle task assembleDebug failed with exit code 1`

**Comment by jeiea** [comment]
I attach my log for record.

<details><summary>iOS build log</summary>

```Launching lib/main.dart on iPhone SE (2nd generation) in debug mode...
Xcode build done.                                           11.2s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **
Xcode's output:
↳
    Invalid depfile: /Users/jeiea/project/.dart_tool/flutter_build/28386ba59aa71afe0f6763b2974e5580/kernel_snapshot.d
    Invalid depfile: /Users/jeiea/project/.dart_tool/flutter_build/28386ba59aa71afe0f6763b2974e5580/kernel_snapshot.d
    Invalid depfile: /Users/jeiea/project/.dart_tool/flutter_build/28386ba59aa71afe0f6763b2974e5580/kernel_snapshot.d
    Invalid depfile: /Users/jeiea/project/.dart_tool/flutter_build/28386ba59aa71afe0f6763b2974e5580/kernel_snapshot.d
    ../../../.asdf/installs/flutter/1.19.0-3.0.pre/.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:11:8: Error: Not found: 'dart:html'
    import 'dart:html';
           ^
    ../../../.asdf/installs/flutter/1.19.0-3.0.pre/.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:28:10: Error: Type 'HttpRequest' not found.
      Future<HttpRequest> _getString(String url, HttpRequest xhr) {
             ^^^^^^^^^^^
    ../../../.asdf/installs/flutter/1.19.0-3.0.pre/.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:28:46: Error: Type 'HttpRequest' not found.
      Future<HttpRequest> _getString(String url, HttpRequest xhr) {
                                                 ^^^^^^^^^^^
    ../../../.asdf/installs/flutter/1.19.0-3.0.pre/.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:20:19: Error: The method 'HttpRequest' isn't defined for the class 'HttpRequestDataReader'.
     - 'HttpRequestDataReader' is from 'package:intl/src/http_request_data_reader.dart' ('../../../.asdf/installs/flutter/1.19.0-3.0.pre/.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart').
    Try correcting the name to the name of an existing method, or defining a method named 'HttpRequest'.
        var request = HttpRequest();
                      ^^^^^^^^^^^
    ../../../.asdf/installs/flutter/1.19.0-3.0.pre/.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:28:46: Error: 'HttpRequest' isn't a type.
      Future<HttpRequest> _getString(String url, HttpRequest xhr) {
                                                 ^^^^^^^^^^^
    ../../../.asdf/installs/flutter/1.19.0-3.0.pre/.pub-cache/hosted/pub.dartlang.org/intl-0.16.1/lib/src/http_request_data_reader.dart:29:31: Error: 'HttpRequest' isn't a type.
        var completer = Completer<HttpRequest>();
                                  ^^^^^^^^^^^
    ../../../.asdf/installs/flutter/1.19.0-3.0.pre/packages/flutter/lib/src/painting/_network_image_web.dart:89:12: Error: Method not found: 'webOnlyInstantiateImageCodecFromUrl'.
        return ui.webOnlyInstantiateImageCodecFromUrl(resolved, // ignore: undefined_function
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    Unhandled exception:
    FileSystemException(uri=org-dartlang-untranslatable-uri:dart%3Ahtml; message=StandardFileSystem only supports file:* and data:* URIs)
          StandardFileSystem.entityForUri  (package:front_end/src/api_prototype/standard_file_system.dart:33:7)
          asFileUri  (package:vm/kernel_front_end.dart:613:37)
          writeDepfile  (package:vm/kernel_front_end.dart:807:21)
    <asynchronous suspension>
          FrontendCompiler.compile  (package:frontend_server/frontend_server.dart:555:15)
    <asynchronous suspension>
          _FlutterFrontendCompiler.compile  (package:flutter_frontend_server/server.dart:40:22)
          starter  (package:flutter_frontend_server/server.dart:178:27)
          main  (file:///b/s/w/ir/cache/builder/src/flutter/flutter_frontend_server/bin/starter.dart:8:30)
          _startIsolate.<anonymous closure>  (dart:isolate-patch/isolate_patch.dart:299:32)
          _RawReceivePortImpl._handleMessage  (dart:isolate-patch/isolate_patch.dart:168:12)
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: Capabilities for Signing & Capabilities may not function correctly because its entitlements use a placeholder team ID. To resolve this, select a development team in the Runner editor. (in target 'Runner' from project 'Runner')
Could not build the application for the simulator.
Error launching application on iPhone SE (2nd generation).
Exited (sigterm)
```
</details>

<details><summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 1.19.0-6.0.pre.27, on Mac OS X 10.15.5 19F101, locale en-KR)
    • Flutter version 1.19.0-6.0.pre.27 at /Users/jeiea/.asdf/installs/flutter/1.19.0-3.0.pre
    • Framework revision 5e3e5a2a1a (25 hours ago), 2020-06-08 17:38:23 -0700
    • Engine revision 580ab9c3c5
    • Dart version 2.9.0 (build 2.9.0-13.0.dev df435c9235)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0-rc4)
    • Android SDK at /Users/jeiea/Library/Android/sdk
    • Platform android-29, build-tools 30.0.0-rc4
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.3

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.45.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.11.0

[✓] Connected device (2 available)
    • iPhone SE (2nd generation) • B29C9E0E-247A-4014-8A0E-54F9CCCE00CD     • ios • com.apple.CoreSimulator.SimRuntime.iOS-13-5 (simulator)

• No issues found!
```
</details>

It's sad that I can't use intl on mobile.

**Comment by jeiea** [comment]
Did you use `import 'package:intl/date_symbol_data_http_request.dart';`? I solved this error with`import 'package:intl/date_symbol_data_local.dart';`.

If you want to support both web and mobile then it may depend on [another issue](https://github.com/flutter/flutter/issues/56829).

**Comment by PsyOhm23** [comment]
I've encounter the same issue and importing the right intl dart file fix it :)
Thank you @jeiea  

BTW, I think this issue may be closed

**Comment by premedios** [comment]
I tried all the above steps and still getting the error! What's going on!?

**Comment by premedios** [comment]
I decided to do the date formatting manually. I'm not recommending Fluttter/Dart to anyone!

**Comment by devnullpointer** [comment]
This is ridiculous, out of the blue i'm getting this error now.. none of my dependencies have changed, using dart 2.14 and flutter 2.5.3

**Comment by mosuem** [comment]
@devnullpointer , @premedios , could you add more details on the errors you are getting and if the above [comment](https://github.com/dart-lang/intl/issues/286#issuecomment-782934999) fixes them?

**Comment by premedios** [comment]
I don't remember what I was doing when I got that error.

**Comment by mosuem** [comment]
Thanks for the quick feedback on this old issue! I will close it then.

**Comment by SijuKJ** [comment]
I had the same issue, one unwanted import was in my code. This import was the culprit "_import 'package:intl/date_symbol_data_http_request.dart'"_. When I removed this import and got worked :)

**Comment by e200** [comment]
> Did you use `import 'package:intl/date_symbol_data_http_request.dart';`? I solved this error with`import 'package:intl/date_symbol_data_local.dart';`.
> 
> If you want to support both web and mobile then it may depend on [another issue](https://github.com/flutter/flutter/issues/56829).

This should be in the docs.
